### PR TITLE
fix: use mcr.microsoft.com for nvidia in mooncake

### LINKED ISSUE #5034: Image pull error for k8s-device-plugin in mooncake
**Describe the bug**

When adding gpu node pool into mooncake, we got image pull error for nvidia `k8s-device-plugin`

``` 
Normal  BackOff  2m43s (x436 over 102m)  kubelet  Back-off pulling image "dockerhub.azk8s.cn/nvidia/oss/nvidia/k8s-device-plugin:1.0.0-beta6"
```

The image path in global cloud is `mcr.microsoft.com/oss/nvidia/k8s-device-plugin:1.0.0-beta6`

**Steps To Reproduce**

**Expected behavior**

**AKS Engine version**

**Kubernetes version**

1.21.4

**Additional context**

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR updates the mooncake configuration for nvidia driver image pull downloads to use the default `mcr.microsoft.com` container image registry.

**Issue Fixed**:

Fixes #5034 

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
cc @Michael-Sinz @bridgetkromhout

File: pkg/api/azenvtypes.go [hunk 1]
```diff
@@ -268,7 +268,7 @@ var (
 		KubernetesSpecConfig: KubernetesSpecConfig{
 			KubernetesImageBase:    "gcr.azk8s.cn/google_containers/",
 			TillerImageBase:        "mcr.microsoft.com/",
-			NVIDIAImageBase:        "dockerhub.azk8s.cn/nvidia/",
+			NVIDIAImageBase:        "mcr.microsoft.com/",
 			AzureCNIImageBase:      "mcr.azk8s.cn/containernetworking/",
```
Comment (Michael-Sinz) [inline pkg/api/azenvtypes.go:272]: What is mcr.azk8s.cn?  (Just asking, we have not run into a problem with it.

File: pkg/api/azenvtypes.go [hunk 2]
```diff
@@ -268,7 +268,7 @@ var (
 		KubernetesSpecConfig: KubernetesSpecConfig{
 			KubernetesImageBase:    "gcr.azk8s.cn/google_containers/",
 			TillerImageBase:        "mcr.microsoft.com/",
-			NVIDIAImageBase:        "dockerhub.azk8s.cn/nvidia/",
+			NVIDIAImageBase:        "mcr.microsoft.com/",
```
Comment (Michael-Sinz) [inline pkg/api/azenvtypes.go:271]: This effectively does what we did to our VMs to get things working.

**Review by Michael-Sinz — APPROVED** [review]

File: pkg/api/azenvtypes.go (hunk 1 above)
Comment (jackfrancis) [reply pkg/api/azenvtypes.go:272]: I don't know for sure, I would guess it is a (no longer maintained?) mirror of certain mcr.microsoft.com k8s images?

**Pull request merged**